### PR TITLE
Task/service cache

### PR DESCRIPTION
**Pull request by dmoranj**
Implements #2

### CODE REVIEW

**Comment by gtorodelvalle** [comment]
LGTM

File: lib/errors.js [hunk 1]
```diff
@@ -27,5 +27,22 @@ module.exports = {
     BadConfiguration: function(msg) {
         this.name = 'BAD_CONFIGURATION';
         this.message = 'The application startup failed due to a bad configuration:' + msg;
+    },
+    DeviceGroupNotFound: function(fields, values) {
+        this.name = 'DEVICE_GROUP_NOT_FOUND';
+        if (values && fields) {
+            this.message = 'Couldn\t find device group for fields: ' + fields + ' and values: ' + values;
```
Comment (AlvaroVega) [inline lib/errors.js]: Maybe Couldn\'t instead of Couldn\t ?

**Comment by AlvaroVega** [comment]
LGTM

**Comment by XavierVal** [comment]
LGTM

**Pull request merged**